XrdAdaptor: improve exception messages

makortel opened the pull request:
#### PR description:

This PR extends the XrdAdaptor exception messages by
* including information on inactive sources to the exceptions that include information on active and disabled sources. 
* the full new URL to the exceptions thrown from `RequestManager::initialize()`
  * the full new URL was already part of the exceptions thrown from `RequestManager::OpenHandler::open()`

These could be useful to investigate the logs of situations such as https://github.com/cms-sw/cmssw/issues/43162

Resolves https://github.com/cms-sw/framework-team/issues/1237

#### PR validation:

Code compiles

#### If this PR is a backport please specify the original PR and why you need to backport that PR. If this PR will be backported please specify to which release cycle the backport is meant for:

To be backported to 15_0_X

makortel committed:
Include information on inactive sources to exception messages

cmsbuild commented:
cms-bot internal usage<!-- bot cache: {"commits":{"259fde05f6182831939f56c6696c6a450b0903b8":{"files":["Utilities/XrdAdaptor/src/XrdRequestManager.cc","Utilities/XrdAdaptor/src/XrdRequestManager.h"],"squashed":false,"time":1740578115}},"emoji":{},"last_seen_sha":"259fde05f6182831939f56c6696c6a450b0903b8","signatures":{}} -->

cmsbuild commented:
A new Pull Request was created by  @makortel for master.

It involves the following packages:

- Utilities/XrdAdaptor (**core**)


@Dr15Jones, @cmsbuild, @makortel, @smuzaffar can you please review it and eventually sign? Thanks.
@wddgit this is something you requested to watch as well.
@antoniovilela, @mandrenguyen, @rappoccio, @sextonkennedy you are the release manager for this.

cms-bot commands are listed <a href="http://cms-sw.github.io/cms-bot-cmssw-cmds.html">here</a>

makortel committed:
Include full new URL in the exception messages from initialize()

The full new URL is already part of the exception message in OpenHandler::open()

cmsbuild commented:
+code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-47462/43894

cmsbuild commented:
Pull request #47462 was updated. @Dr15Jones, @cmsbuild, @makortel, @smuzaffar can you please check and sign again.

makortel commented:
FYI @bbockelm @dynamic-entropy

makortel commented:
Comparison differences are related to https://github.com/cms-sw/cmssw/issues/47071

makortel commented:
+core

cmsbuild commented:
This pull request is fully signed and it will be integrated in one of the next master IBs (tests are also fine). This pull request will now be reviewed by the release team before it's merged. @sextonkennedy, @antoniovilela, @rappoccio, @mandrenguyen (and backports should be raised in the release meeting by the corresponding L2)

mandrenguyen commented:
+1
